components/calendar/src/hebrew.rs

```diff
@@ -64,17 +64,16 @@ const SHEVAT: u8 = 11;
 const ADAR: u8 = 12;
 const ADARII: u8 = 13;
 
-/// The inner date type used for representing [`Date`]s of [`Hebrew`]. See [`Date`] and [`Hebrew`] for more details.
+/// The inner date type used for representing [`Date`]s of [`BookHebrew`]. See [`Date`] and [`BookHebrew`] for more details.
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
-pub struct HebrewDateInner(ArithmeticDate<Hebrew>);
-
-impl CalendarArithmetic for Hebrew {
+pub struct BookHebrewDateInner(ArithmeticDate<BookHebrew>);
```

Comment:
Note: We shouldn't use ArithmeticDate for BookHebrew; we should fix this in a follow-up.

Reply:
Yeah, sorry, I think there may have been a misunderstanding:

I was suggesting `BookHebrewDate` be a completely internal abstraction that we use to do the hebrew calendar algorithms. We still will only have Hebrew and HebrewDateInner (which wraps an ArithmeticDate as usual).

When we are using the book algorithms, we quickly convert between HebrewDateInner and BookHebrewDate before calling the relevant methods

Reply:
Also: I don't think we are fully clear on the actual existence of a `BookHebrew` calendar. The month numbering is a thing that was used in ancient times, but in those times the year numbering was different _anyway_. The _Anno Mundi_ tradition was not there from the get-go. So this calendar represents something that was never a real calendar.


What's going on is that the canonical way to number these months is not ordinally, since Nisan is considered the start of the year for festivals and such. But nobody numbers years that way.

The book decided to use the canonical numbering rather than the ordinal numbering, which is fine, but as far as I can tell there is no actual calendar that was ever used corresponding to that numbering that also numbers years the same way.